app/Http/Requests/PostRequest.php

```diff
@@ -26,12 +26,12 @@ public function rules()
         }
 
         return [
-            'campaign_id' => 'required|integer',
+            'campaign_id' => 'required_without:action_id|integer',
             'northstar_id' => 'nullable|objectid',
-            'type' => 'required|string|in:photo,voter-reg,text,share-social',
+            'type' => 'required_without:action_id|string|in:photo,voter-reg,text,share-social',
```

Comment:
Will this make our conditional validation rules more complicated (since we'd need to make a database query to determine what kind of a payload to expect for a given `action_id`)? Thinking about how `file` (should) be required only for `photo` type posts, and we [allow different statuses](https://github.com/DoSomething/rogue/blob/4a12be9f680f7cec776be1e81573357440a0cd5a/app/Http/Requests/PostRequest.php#L66-L77) based on the type of post as well.

Reply:
Sorry @DFurnes this PR got a little mixed up bc I accidentally branched from this - but these changes were intended for updates in #837 ! 

I'm going to move the discussion there!